### PR TITLE
Display translated text in campaign dialog

### PR DESCRIPTION
**Pull request by dimag0g**
The root cause of the issue is explained in Campaign data doesn't get translated. #3807.
Here are the screenshots before the change:

homm2_ru

and after the change:

homm2_ru_new

### CODE REVIEW

File: src/fheroes2/campaign/campaign_data.cpp [hunk 1]
```diff
@@ -228,46 +228,47 @@ namespace
         std::vector<Campaign::ScenarioData> scenarioDatas;
         scenarioDatas.reserve( 10 );
 
-        scenarioDatas.emplace_back( 0, std::vector<int>{ 1 }, Campaign::ScenarioBonusData::getCampaignBonusData( 0, 0 ), "CAMPG01.H2C", rolandCampaignScenarioNames[0],
-                                    rolandCampaignDescription[0],
+        scenarioDatas.emplace_back( 0, std::vector<int>{ 1 }, Campaign::ScenarioBonusData::getCampaignBonusData( 0, 0 ), "CAMPG01.H2C",
+                                    _( rolandCampaignScenarioNames[0] ), _( rolandCampaignDescription[0] ),
```
Comment (undef21) [inline src/fheroes2/campaign/campaign_data.cpp:232]: Usually gettext is used with string literals.
What are you thinking about fheroes2's approach (replace global arrays with local)?
- replace this with something likes `GetDescription(index)`;
- implement 
`GetDescription(index){
char* descriptions = { _("descr1"), ... };
return descriptions(index);
}`
You could see this for example [here](https://github.com/ihhub/fheroes2/blob/ffdb801bb3a371bfc6a45ce609883d80f7a623d9/src/fheroes2/campaign/campaign_scenariodata.cpp#L342) and [here](https://github.com/ihhub/fheroes2/blob/ffdb801bb3a371bfc6a45ce609883d80f7a623d9/src/fheroes2/castle/castle.cpp#L654)

**Comment by dimag0g** [comment]
@undef21 Good idea! Please take a look at this new commit.

**Comment by dimag0g** [comment]
@ihhub Could you please take a look at this PR? I think it's ready to be merged.

**Comment by ihhub** [comment]
@dimag0g , thank you very much for this fix!

**Pull request merged**